### PR TITLE
Trivial fixes for MP3Filter

### PR DESCRIPTION
**Pull request by bertm**
This fixes a few header parsing and consistency bugs in the MP3Filter, while introducing support for dropping entire ID3v1 and ID3v2 tags.

I tested this with a random collection of music files on my PC, as well as @ArneBab's reported broken file, and all were filtered well where they were previously all broken.

### CODE REVIEW

**Comment by ArneBab** [comment]
I don’t understand mp3 well enough to really judge the pull-request, but it looks good, and it definitely shouldn’t make anything worse.

I plan to test this in a bit more than a week.

File: src/freenet/client/filter/MP3Filter.java [hunk 1]
```diff
@@ -118,9 +118,12 @@ public void filter(InputStream input, OutputStream output) throws DataFilterExce
...
 				}
-				byte samplerateIndex = (byte) ((frameHeader & 0x0000c0000) >>> 10); //2 bits
-				if(samplerateIndex == 3) continue; // Not valid
-				boolean paddingBit = ((frameHeader & 0x00000300) >>> 9) == 1 ? true : false;
+				byte samplerateIndex = (byte) ((frameHeader & 0x00000c00) >>> 10); //2 bits
+				if (samplerateIndex == 3) {
+					foundStream = false;
+					continue; // Not valid
+				}
+				boolean paddingBit = ((frameHeader & 0x00000200) >>> 9) == 1 ? true : false;
```
Comment (Bombe) [inline src/freenet/client/filter/MP3Filter.java]: `? true : false`? Seriously? Please remove that. :)

File: src/freenet/client/filter/MP3Filter.java [hunk 2]
```diff
@@ -167,6 +170,29 @@ public void filter(InputStream input, OutputStream output) throws DataFilterExce
...
+				int size = 0;
+				size |= (encodedSize[0] & 0x7F) << 21;
+				size |= (encodedSize[1] & 0x7F) << 14;
+				size |= (encodedSize[2] & 0x7F) << 7;
+				size |= (encodedSize[3] & 0x7F);
+				in.skip(size);
+				Logger.normal(this, "Skipped " + size + " bytes of ID3v2 data");
+				frameHeader = in.readInt();
+				foundStream = (frameHeader & 0xffe00000) == 0xffe00000;
+			} else if (!foundStream && (frameHeader & 0xffffff00) == 0x54414700) {
```
Comment (Bombe) [inline src/freenet/client/filter/MP3Filter.java]: ID3v1 headers are not located at the beginning of the file. Am I missing something here?

**Comment by Bombe** [comment]
The ID3 tag skipping looks good, too. Now the only thing missing is some refactoring and a test. :)

**Comment by ArneBab** [comment]
Feel free to use my file. The song is a traditional and it’s sung by me.

However 500kiB could be a bit too large…

**Comment by Bombe** [comment]
Uhm, encode a second of silence and tag that. No need for more than that.

**Comment by bertm** [comment]
I'm working on it.
Op 12 jan. 2016 19:31 schreef "David Roden" <notifications@github.com>:

**Comment by Bombe** [comment]
@bertm awesome!

**Comment by ArneBab** [comment]
Here’s a one second file with a dieing sine sound: CHK@x8~GoW0UAZ9PR8W44XTFN3-U6IWET3DP3yHqwylsGFI,m62xSWGyvgSviCj562RXB7exZTcPXgDBEjNzXeX5zuQ,AAMC--8/dieing-sine-48khz.mp3

(no silence to ensure that we *hear* it when the filter breaks)

**Comment by ArneBab** [comment]
@bertm will you finish this yourself? I’m working on an m3u filter right now #510 so along with #429 we’d have full support for radio over Freenet.

**Comment by bertm** [comment]
Yep, I have something almost commit-ready laying around. My thesis sure is a killer to my available time, but I'll try and have a final look at it tonight.

Thanks for your work on an M3U filter by the way!

**Comment by ArneBab** [comment]
thank you!

**Comment by bertm** [comment]
I added several test instances of 200ms worth of generated audio data, encoded in various ways. I did not choose to use silence, for that would reduce the variability of the MPEG frames to approximately zero.

Implementing this I encountered and fixed yet another bug (this time caused by partial or even erroneous online documentation of the peculiarities of frame length).

**Comment by bertm** [comment]
The integration test failure is completely unrelated; one of the tests ran out of heap space in some splitfile test.

**Pull request merged**